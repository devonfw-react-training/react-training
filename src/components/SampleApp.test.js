import React from "react";
import ReactDOM from "react-dom";
import SampleApp from "./SampleApp";

import { shallow, mount, render as erender } from "enzyme";
import { render, cleanup, fireEvent} from 'react-testing-library';

describe("renders without crashing", () => {
  it("when default approach used", () => {
    const div = document.createElement("div");
    ReactDOM.render(<SampleApp />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  describe("when enzyme", () => {
    it("shallow used", () => {
      shallow(<SampleApp />);
    });

    it("mount used", () => {
      mount(<SampleApp />);
    });
  });

  describe("when snapshots taken with enzyme", () => {
    it("shallow", () => {
      expect(shallow(<SampleApp />)).toMatchSnapshot();
    });

    it("mount", () => {
      expect(mount(<SampleApp />)).toMatchSnapshot();
    });

    it("render", () => {
      expect(erender(<SampleApp />)).toMatchSnapshot();
    });
  });
});

describe("renders welcome message checked with enzyme", () => {
  const wrapper = shallow(<SampleApp />);
  const welcome = <h1 className="App-title">Welcome to React</h1>;

  it('checked with plain enzyme', () => {
    expect(wrapper.contains(welcome)).toEqual(true);
  });

  it('checked with jest-enzyme', () => {
    expect(wrapper).toContainReact(welcome);
  });
});


describe("renders with react-testing-library", () => {
  afterEach(cleanup);

  it('welcome message checked', () => {
    const {getByText} = render(<SampleApp />)
    const aboutAnchorNode = getByText('Welcome to React');
    expect(aboutAnchorNode).toMatchSnapshot();
  })
});

describe('check more advanced API from react-testing-library ', () => {
  class TestComponent extends React.Component {
    constructor() {
      super()
      this.state = {count: 0}
    }
  
    render() {
      const {count} = this.state
  
      return (
        <button onClick={() => this.setState({count: count + 1})}>
          Click to increase: {count}
        </button>
      )
    }
  }

  it('diff snapshot for state change', () => {
  
    const {getByText, asFragment} = render(<TestComponent />)
    const firstRender = asFragment()
    
    fireEvent.click(getByText(/Click to increase/))
    
    // This will snapshot only the difference between the first render, and the
    // state of the DOM after the click event.
    // See https://github.com/jest-community/snapshot-diff
    expect(firstRender).toMatchDiffSnapshot(asFragment())
  });
});
