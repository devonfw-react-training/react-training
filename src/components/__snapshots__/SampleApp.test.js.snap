// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check more advanced API from react-testing-library  diff snapshot for state change 1`] = `
"Snapshot Diff:
- First value
+ Second value

  <DocumentFragment>
    <button>
-     Click to increase: 0
+     Click to increase: 1
    </button>
  </DocumentFragment>"
`;

exports[`renders with react-testing-library welcome message checked 1`] = `
<h1
  class="App-title"
>
  Welcome to React
</h1>
`;

exports[`renders without crashing when snapshots taken with enzyme mount 1`] = `
<SampleApp>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
    </header>
    <p
      className="App-intro"
    >
      To get started, edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
  </div>
</SampleApp>
`;

exports[`renders without crashing when snapshots taken with enzyme render 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <img
      alt="logo"
      class="App-logo"
      src="logo.svg"
    />
    <h1
      class="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    class="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;

exports[`renders without crashing when snapshots taken with enzyme shallow 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
